transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -457,6 +457,26 @@ public final ChannelHandler remove(String name) {
         return (T) remove(getContextOrDie(handlerType)).handler();
     }
 
+    public final <T extends ChannelHandler> T removeIfExists(String name) {
+        return removeIfExists(context(name));
+    }
+
+    public final <T extends ChannelHandler> T removeIfExists(Class<T> handlerType) {
+        return removeIfExists(context(handlerType));
+    }
+
+    public final <T extends ChannelHandler> T removeIfExists(ChannelHandler handler) {
+        return removeIfExists(context(handler));
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T extends ChannelHandler> T removeIfExists(ChannelHandlerContext ctx) {
+        if (ctx == null) {
+            return null;
+        }
+        return (T) remove((AbstractChannelHandlerContext) ctx).handler();
```

Comment:
@doom369 I think this is not safe as in the meantime it could have been removed from another thread which would have this method throw a `NoSuchElementException`

Reply:
@normanmaurer I don't see any place where NoSuchElementException could be thrown in above code.

Reply:
ah never mind... this should work...